src/state/entities/components/triggers.ts

```diff
@@ -1,12 +1,73 @@
 import entitiesStateManager from 'state/entities/entitiesStateManager';
 import { ASYNC_ENTITY_KEYS } from 'models/state/entities.models';
 import { handleComponent, triggerFlashMessage } from 'state/ui/actions';
-import { IComponentEntity } from 'models/state/components.model';
+import {
+    IComponent,
+    IComponentByNameAndVersionPayload,
+    IFetchComponentsListPayload,
+} from 'models/state/components.model';
+import { StateChangeNotification } from 'models/state.models';
 
-export const triggerUpdateComponent = (payload: IComponentEntity | IComponentEntity[], bulk?: boolean) =>
+export const triggerFetchComponents = (payload: IFetchComponentsListPayload) =>
+    entitiesStateManager.triggerAsyncEntityFetch<{}>({
+        asyncEntityToFetch: {
+            asyncEntityKey: ASYNC_ENTITY_KEYS.components,
+            refreshMode: 'always',
+            resetDataOnTrigger: false,
+        },
+        extraInputSelector: () => payload,
+        notificationsToTrigger: [StateChangeNotification.DESIGN_COMPONENTS_LIST],
+    });
+export const triggerFetchComponentDetail = (payload: IComponentByNameAndVersionPayload) =>
+    entitiesStateManager.triggerAsyncEntityFetch<{}>({
+        asyncEntityToFetch: {
+            asyncEntityKey: ASYNC_ENTITY_KEYS.componentDetail,
+            refreshMode: 'always',
+            resetDataOnTrigger: true,
+        },
+        extraInputSelector: () => payload,
+        notificationsToTrigger: [StateChangeNotification.DESIGN_COMPONENT_DETAIL],
+    });
+
+export const triggerCreateComponentDetail = (payload: IComponent) =>
+    entitiesStateManager.triggerAsyncEntityCreate<{}>({
+        asyncEntityToCreate: {
+            asyncEntityKey: ASYNC_ENTITY_KEYS.componentDetail,
+        },
+        extraInputSelector: () => payload,
+        notificationsToTrigger: [StateChangeNotification.DESIGN_COMPONENT_DETAIL],
+    });
+
+export const triggerUpdateComponentDetail = (payload: IComponent) =>
     entitiesStateManager.triggerAsyncEntityUpdate<{}>({
         asyncEntityToUpdate: {
-            asyncEntityKey: ASYNC_ENTITY_KEYS.components,
+            asyncEntityKey: ASYNC_ENTITY_KEYS.componentDetail,
+        },
+        extraInputSelector: () => payload,
+        notificationsToTrigger: [StateChangeNotification.DESIGN_COMPONENT_DETAIL],
+        onSuccess: ({ dispatch }) => {
+            dispatch(triggerFlashMessage({
+                translationKey: 'flash_messages.component.edit',
+                type: 'success',
+            }));
+        },
```

Comment:
is this also done for scripts?

Reply:
No, I though that it is better to  display notifications when users create or update a component.

Reply:
okay, could you also add this for scripts in a next issue? or add it to the backlog?